Makefile

```diff
@@ -84,35 +84,48 @@ setup: pilot/platform/kube/config
 #-----------------------------------------------------------------------------
 # Target: depend
 #-----------------------------------------------------------------------------
-.PHONY: depend 
+.PHONY: depend dep.update
 .PHONY: depend.status depend.ensure depend.graph
 
-depend: depend.ensure
+# Pull depdendencies, based on the checked in Gopkg.lock file.
+# Developers must manually call dep.update if adding new deps or to pull recent
+# changes.
+depend: vendor
+
+# Target to update the Gopkg.lock with latest versions.
+# Should be run when adding any new dependency and periodically.
+depend.update: ${GOPATH}/bin/dep; $(info $(H) ensuring dependencies are up to date...)
+	dep ensure
+	dep ensure -update
```

Comment:
why two? isnt dep ensure sufficient?

Reply:
`dep ensure -update` can't run without the lock file, which is generated by `dep ensure`.

Reply:
Actually this change was started before we decided to checkin the lock file, no longer needed.